### ISSUE TITLE
fractional order Laguerre gives wrong result

### ISSUE THREAD

**Issue by cbm755** [sympy/sympy#10961]
```
In [22]: laguerre(2, x)            # yes
Out[22]: x**2/2 - 2*x + 1

In [23]: laguerre(n, x)            # yes
Out[23]: laguerre(n, x)

In [24]: laguerre(pi, x)           # ok
Out[24]: laguerre(pi, x)

In [25]: laguerre(S(5)/2, x)       # silently, incorrectly gives same as n=2
Out[25]: x**2/2 - 2*x + 1

In [26]: laguerre(-1, x)           # correctly raises an exception
Out[25]: x**2/2 - 2*x + 1
```

Perhaps an `int()` cast somewhere which could be more precise?  It should either stay unevaluated or raise an exception, TBD.

"Easy to Fix" tag is a guess, not a promise ;-)

**Comment by zanzibar7** [comment]
I think the issue actually comes about because `laguerre` is calling `laguerre_poly` directly,
rather than through `_ortho_poly` like the other standard orthogonal polynomials.

**Comment by Lakshya1605** [comment]
I would like to work on this. It might take a little time, though. I am new here.

**Comment by cbm755** [comment]
I think @zanzibar7 has a solution up above: perhaps you could start by reviewing his work?

@zanzibar7 is there a formal PR?

**Comment by zanzibar7** [comment]
I did not submit a pull request on this.  

@Lakshya1605 , it would be great if you wanted to finish this off.

**Comment by Lakshya1605** [comment]
Please can some one elaborate on what is expected

**Comment by zanzibar7** [comment]
In the comments above, from April, skirpichev and I suggested the changes that needed to be made and tested so we don't have this issue with fractional powers for Laguerre polynomials.  Read over those changes, check that they work, make the sure the test-suite passes, and if so, put them into a pull request we can review, comment on, and eventually merge.

**Comment by Lakshya1605** [comment]
@zanzibar7    Can you please tell me what is the use of  _ortho_poly = staticmethod(laguerre_poly) and why did you use this in  your commit,I was thinking what is wrong by directly returning laguerre_poly(int(n), x, 0).Sorry this may seem a very stupid question but i am new in here and not properly familiar with the code.Please help me

**Comment by zanzibar7** [comment]
All of the classical families of orthogonal polynomials like Laguerre  polynomials are subclassed from `OrthogonalPolynomial`.  `OrthogonalPolynomial` is  intended for housekeeping of stuff that's common, like dealing with fractional  exponents -- we're supposed to call members of the superclass like `_eval_at_order` when we need to evaluate the polynomial.

In order for to work, `laguerre` has to have a member function called 
`_ortho_poly` that `_eval_at_order` will call.  The line `_ortho_poly = staticmethod(laguerre_poly) ` is creating that member (the nasty details of `laguerre_poly` is defined elsewhere and imported here)    Once that's set up, we can have `eval()` call `cls._eval_at_order(n, x)` like in all the others polynomial classes and trust it to take care of the fractional exponent error handling.

We don't see classes set up like this very often, so don't worry about the details of why too much.  At the simplest level, all the other classes like `hermite` and `legendre` are set up this way, so we want to make sure `laguerre` is set up the same.

Aside:  use backquotes to hilight code in your comments.

**Comment by Lakshya1605** [comment]
Thank you @zanzibar7 . Working on this issue.

**Comment by Lakshya1605** [comment]
I have submitted a pull request but i am getting " continuous-integration/travis-ci/pr — The Travis CI build failed "  please help me  @cbm755  @zanzibar7 .I am not pretty sure why this error is coming

**Comment by zanzibar7** [comment]
When you submit a pull request to fix an issue, include the issue number ( #10961 ) in the request.  That way, a link to it will show up here.

I'm guessing we are talking about pull request  #11698.

Looks like you have a simple spelling error.  This would have been caught if you ran the test suite before submitting.  If you're in the terminal, working from the sympy home directory, run 
```
bin/test spec_polynomials
```
to test the changes you made to `sympy/functions/special/tests/test_spec_polynomials.py` and 
`sympy/functions/special/polynomials.py`.  If that passes without errors, run `bin/test` to run the full test suite (which takes a long time to run).  If that passes and `bin/doctest` also passes, you're ready to submit a pull request.

There's allot of information on this in the [development workflow](http://github.com/sympy/sympy/wiki/Development-workflow) documentation.

**Comment by Lakshya1605** [comment]
@zanzibar7  I am ran the command ./bin/test but i am getting test failed in "sympy/printing/tests/test_theanocode.py[27]" &"sympy/plotting/intervalmath/tests/test_interval_functions.py[24]"&"sympy/plotting/intervalmath/tests/test_intervalmath.py[7]" Though i have changed none of the this files Please help me.

**Comment by Lakshya1605** [comment]
Thank you @zanzibar7  for your help .Pull request sent #11698

**Comment by namannimmo10** [comment]
The issue can be closed now. PR was merged.

**Comment by Smit-create** [comment]
@oscarbenjamin  Can this be closed as I think it is fixed in #11698 ?

**Comment by smichr** [comment]
It is tested there. This can be closed.
